docs/custom-domain/registry-setup.md

```diff
@@ -0,0 +1,103 @@
+  
+## Prepare your existing registry
+We will enable two features on your registry that are currently in preview:
+- Data Endpoints:\
+  This feature provides a dedicated endpoint for downloading content from your registry. If you have a registry in East US, on enabling this feature, a data endpoint is automatically created for you: `eastus.data.myregistry.azurecr.io`
+  
+- ACR Managed Identities:\
+  Managed Identities provide a mechanism to associate an Azure Active Directory identity with your registry, while relieving you of the burden of managing credentials. To learn more, see the documentation [here](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview).\
+ ACR supports both user assigned and system assigned managed identities.
+
+### Enable preview features
+1. `az login`
+2. `az account set -s <subscription-id-or-name> `
+3. `apiVersion="api-version=2019-12-01-preview"`
+4. `mgmtEndpoint=$(az cloud show --query endpoints.resourceManager -o tsv)`
+5. `accessToken=$(az account get-access-token --query accessToken -o tsv)`
+6. `registryResourceId=$(az acr show -n myregistry --query id -o tsv)`
+7. You can either enable a system assigned managed identity, a user assigned managed identity, or both for your registry. We recommend using system assigned managed identity to enable advanced scenarios with virtual networks that, although not supported currently, are [coming soon](#enhanced-security-with-virtual-networks). Do _one_ of the following:
+   - To enable only system assigned managed identity: 
+     - `identity="{\"type\": \"systemAssigned\"}"`
+   - To enable user assigned managed identity, with or without a system identity: 
+     - Create a user assigned managed identity following the instructions [here](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal).
+     - Do _one_ of the following:
+       - To enable _only_ user assigned managed identity:
+         - `userMIResourceId=$(az identity show -n <user-mi-name> -g <user-mi-resource-group> --subscription <user-mi-subscription-id-or-name> --query id -o tsv)`
+         - `identity="{\"type\": \"userAssigned\", \"userAssignedIdentities\": {\"${userMIResourceId}\":{}}}"`
+       - To enable _both_ user and system assigned managed identities:
+         - `userMIResourceId=$(az identity show -n <user-mi-name> -g <user-mi-resource-group> --subscription <user-mi-subscription-id-or-name> --query id -o tsv)`
+         - `identity="{\"type\": \"systemAssigned,userAssigned\", \"userAssignedIdentities\": {\"${userMIResourceId}\":{}}}"`
+8. `payload="{\"identity\": $identity, \"properties\": {\"dataEndpointEnabled\": true}}"`
+9. `endpoint="$mgmtEndpoint$registryResourceId?$apiVersion"`
+10. `curl -f -X PATCH -H "Authorization: Bearer $accessToken" -H "Content-Type: application/json" $endpoint -d "$payload"`
```

Comment:
please use "az acr identity assign" to replace the the "curl"

Reply:
In addition to enabling identity, it is also enable data endpoints.

Reply:
I see. In the latest CLI, you can use "az acr update --data-endpoint-enabled" to enable data endpoint.

Reply:
Thanks, updated!